src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2267,7 +2267,15 @@ internal void AdjustScale(int digits, bool fRound)
                 throw new OverflowException(SQLResource.ArithOverflowMessage);
 
             bNewScale = (byte)(lAdjust + _bScale);
-            bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+
+            if (Value == Decimal.Zero && _bScale == 0 && lAdjust > 0)
+            {
+                bNewPrec = bNewScale;
```

Comment:
`Value == Decimal.Zero && _bScale == 0` indicates SqlDecimal 0 (flat zero), which has prevision 1 and scale 0. Not any other case satisfies this condition. 
`bNewScale` value already checked and verified to be greater than or equal to 0, and less than or equal to s_NUMERIC_MAX_PRECISION in previous lines. So `bNewPrec` does not needs to be checked since it has the same value as `bNewScale`